wazo_agentd/service/handler/extension_status.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2023 The Wazo Authors  (see the AUTHORS file)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import logging
+
+from xivo import debug
+from xivo_dao.helpers import db_utils
+
+from wazo_agentd.exception import AgentNotLoggedError
+
+logger = logging.getLogger(__name__)
+
+
+class ExtensionStatusHandler:
+    def __init__(self, agent_status_dao, user_dao, logoff_manager):
+        self._agent_status_dao = agent_status_dao
+        self._user_dao = user_dao
+        self._logoff_manager = logoff_manager
+
+    @debug.trace_duration
+    def handle_on_extension_status_updated(self, msg):
+        if msg['StatusText'] == 'Unavailable':
+            with db_utils.session_scope():
+                try:
+                    user = self._user_dao.get_user_by_number_context(
+                        msg['Exten'], msg['Context']
+                    )
+                    self.handle_logoff_user_agent(
+                        user.uuid, tenant_uuids=[user.tenant_uuid]
+                    )
+                except (LookupError, AgentNotLoggedError) as e:
+                    logger.debug(f'Cannot logout agent ({e})')
```

Comment:
I would separate that into two separate try excepts to avoid triggering unnecessary errors and add more informative logging.
 ```suggestion
               
                try:
                    user = self._user_dao.get_user_by_number_context(
                        msg['Exten'], msg['Context']
                    )
                except LookupError as e:
                    logger.error(f'Cannot retrieve user from extension {msg["Exten"]}@{msg["Context"]}')
                    return
                    
                try:
                    self.handle_logoff_user_agent(
                        user.uuid, tenant_uuids=[user.tenant_uuid]
                    )
                except AgentNotLoggedError as e:
                    logger.debug(f'Cannot logout agent for user {user.uuid} ({e})')
```

Reply:
I assume that not retrieving a user is an error.

Reply:
For info if I disconnect a non-agent user this stack trace is produced:
```
2023-05-04 17:19:14,386 [2223456] (ERROR) (BusConsumer): Handler 'on_extension_status_updated' for event 'ExtensionStatus' failed
May 04 17:19:14 wazo-stack-dev wazo-agentd[2223456]: Traceback (most recent call last):
May 04 17:19:14 wazo-stack-dev wazo-agentd[2223456]:   File "/usr/src/wazo/wazo-agentd/wazo_agentd/service/manager/logoff.py", line 40, in _check_user_has_agent
May 04 17:19:14 wazo-stack-dev wazo-agentd[2223456]:     self._agent_dao.agent_with_user_uuid(
May 04 17:19:14 wazo-stack-dev wazo-agentd[2223456]:   File "/usr/lib/python3/dist-packages/xivo_dao/helpers/db_manager.py", line 78, in wrapped
May 04 17:19:14 wazo-stack-dev wazo-agentd[2223456]:     return func(session, *args, **kwargs)
May 04 17:19:14 wazo-stack-dev wazo-agentd[2223456]:   File "/usr/lib/python3/dist-packages/xivo_dao/agent_dao.py", line 55, in agent_with_user_uuid
May 04 17:19:14 wazo-stack-dev wazo-agentd[2223456]:     raise LookupError(f'no agent found for user {user_uuid}')
May 04 17:19:14 wazo-stack-dev wazo-agentd[2223456]: LookupError: no agent found for user ad5b78cf-6e15-45c7-9ef3-bec36e07e8d6
May 04 17:19:14 wazo-stack-dev wazo-agentd[2223456]: During handling of the above exception, another exception occurred:
May 04 17:19:14 wazo-stack-dev wazo-agentd[2223456]: Traceback (most recent call last):
May 04 17:19:14 wazo-stack-dev wazo-agentd[2223456]:   File "/usr/lib/python3/dist-packages/xivo_bus/mixins.py", line 201, in __dispatch
May 04 17:19:14 wazo-stack-dev wazo-agentd[2223456]:     handler(payload)
May 04 17:19:14 wazo-stack-dev wazo-agentd[2223456]:   File "/usr/src/wazo/wazo-agentd/wazo_agentd/service/proxy.py", line 148, in on_extension_status_updated
May 04 17:19:14 wazo-stack-dev wazo-agentd[2223456]:     return self.extension_status_handler.handle_on_extension_status_updated(
May 04 17:19:14 wazo-stack-dev wazo-agentd[2223456]:   File "/usr/src/wazo/wazo-agentd/wazo_agentd/service/handler/extension_status.py", line 28, in handle_on_extension_status_updated
May 04 17:19:14 wazo-stack-dev wazo-agentd[2223456]:     self.handle_logoff_user_agent(
May 04 17:19:14 wazo-stack-dev wazo-agentd[2223456]:   File "/usr/src/wazo/wazo-agentd/wazo_agentd/service/handler/extension_status.py", line 41, in handle_logoff_user_agent
May 04 17:19:14 wazo-stack-dev wazo-agentd[2223456]:     self._logoff_manager.logoff_user_agent(
May 04 17:19:14 wazo-stack-dev wazo-agentd[2223456]:   File "/usr/src/wazo/wazo-agentd/wazo_agentd/service/manager/logoff.py", line 19, in logoff_user_agent
May 04 17:19:14 wazo-stack-dev wazo-agentd[2223456]:     self._check_user_has_agent(user_uuid, tenant_uuids)
May 04 17:19:14 wazo-stack-dev wazo-agentd[2223456]:   File "/usr/src/wazo/wazo-agentd/wazo_agentd/service/manager/logoff.py", line 44, in _check_user_has_agent
May 04 17:19:14 wazo-stack-dev wazo-agentd[2223456]:     raise NoSuchAgentError()
May 04 17:19:14 wazo-stack-dev wazo-agentd[2223456]: wazo_agentd.exception.NoSuchAgentError
```